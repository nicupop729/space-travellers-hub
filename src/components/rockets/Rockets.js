import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getRockets } from '../../redux/rockets/rockets';
import Profile from '../profile/Profile';
import Rocket from './Rocket';
import './rockets.css';

let loadFirstTime = false;

const Rockets = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rocketsReducer);
  useEffect(() => {
    if (!loadFirstTime) {
      dispatch(getRockets());
      loadFirstTime = true;
    }
  }, [dispatch]);

  Profile();

  return (
    <ul className="Rockets">
      {rockets.map(({
        image, name, description, id, reserved,
      }) => (
        <Rocket
          id={id}
          key={id}
          image={image}
          name={name}
          description={description}
          reserved={reserved}
        />
      ))}
    </ul>
  );
};

export default Rockets;
