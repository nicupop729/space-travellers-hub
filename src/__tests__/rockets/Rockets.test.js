import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Rockets from '../../components/rockets/Rockets';

describe('snapshot testing', () => {
  test('snapshot for Rockets component', () => {
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });
});
