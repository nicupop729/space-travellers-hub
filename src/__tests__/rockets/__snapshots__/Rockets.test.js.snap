// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Rockets component 1`] = `
<ul
  className="Rockets"
/>
`;
